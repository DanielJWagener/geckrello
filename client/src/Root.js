// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
