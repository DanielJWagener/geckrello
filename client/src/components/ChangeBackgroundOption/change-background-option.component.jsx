import React from "react";
import { connect } from "react-redux";

import { updateBoard } from "../../actions";
import colorThemes from "../../utilities/colorThemes";

const ColorPalette = props => {
  const theme = colorThemes[props.color];

  const updateBackgroundColor = () => {
    let root = document.documentElement;

    root.style.setProperty("--color-primary", theme.base);
    root.style.setProperty("--color-primary-lightest", theme.lightest);
    root.style.setProperty("--color-primary-lighter", theme.lighter);
    root.style.setProperty("--color-primary-light", theme.light);
    root.style.setProperty("--color-primary-dark", theme.dark);
    root.style.setProperty("--color-primary-darker", theme.darker);
    root.style.setProperty("--color-primary-darkest", theme.darkest);
    props.updateBoard(props.board._id, { background: props.color });
  };
  return (
    <div
      className="color-palette"
      style={{
        borderRadius: "10px",
        height: "7.5rem",
        width: "14rem",
        backgroundColor: theme.base,
        margin: "1.5rem",
        cursor: "pointer"
      }}
      onClick={updateBackgroundColor}
    ></div>
  );
};

const mapStateToProps = ({ board }) => {
  return { board };
};

export default connect(mapStateToProps, { updateBoard })(ColorPalette);
