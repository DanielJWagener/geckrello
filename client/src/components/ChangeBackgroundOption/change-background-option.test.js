import { shallow } from "enzyme";
import React from "react";
import ChangeBackgroundOption from "./change-background-option.component";
import Root from "../../Root";

it("renders ChangeBackgroundOption component", () => {
  expect(
    shallow(
      <Root>
        <ChangeBackgroundOption />
      </Root>
    )
  ).toMatchSnapshot();
});
