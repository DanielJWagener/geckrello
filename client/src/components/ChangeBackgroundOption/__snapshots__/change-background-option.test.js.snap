// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChangeBackgroundOption component 1`] = `ShallowWrapper {}`;
